app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java

```diff
@@ -105,7 +110,21 @@ protected String requestToken() throws UnableToSendLabOrderException {
         return token;
     }
 
+    protected String retrievePrivateKey() throws SecretRetrievalException {
+        var senderPrivateKey =
+                "report-stream-sender-private-key-" + ApplicationContext.getEnvironment();
+        String key = getCachedPrivateKey();
+        if (key != null) {
+            return key;
+        }
+
+        key = secrets.getKey(senderPrivateKey);
+        setCachedPrivateKey(key);
```

Comment:
Initially I was confused why setCachedPrivateKey was being called for each execution. It's not of course -- I just missed the earlier return.

Would it be helpful to consolidate this into a single return statement? For example:

```
String key = getCachedPrivateKey();

if (key == null) {
    key = secrets.getKey(senderPrivateKey);
    setCachedPrivateKey(key);
}

return key;
```

Reply:
We discussed this syntax and we decided to keep the original because it returns earlier.